### ISSUE TITLE
Fix false positives for maps in property-no-unknown

### ISSUE THREAD

**Issue by MrVladevoit** [stylelint/stylelint#4834]
> Clearly describe the bug

Rule **property-no-unknown** highlight syntax inside [LESS Maps](http://lesscss.org/features/#maps-feature). I don't understand why. 

[Screenshot with error](https://i.postimg.cc/J4zJNbP4/2020-06-19-10-04-49.png)

> Which rule, if any, is the bug related to?

`property-no-unknown `

> What code is needed to reproduce the bug?

```less
@sizes: {
  mobile: 320px; // -> Unexpected unknown property "mobile"
  tablet: 768px; // -> Unexpected unknown property "tablet"
  desktop: 1024px; // -> Unexpected unknown property "desktop"
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "property-no-unknown": "true"
  }
}
```

> Which version of stylelint are you using?

`13.6.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

"Webstorm plugin"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

 "Yes, it's related to LESS maps."

> What did you expect to happen?

"No warnings inside LESS maps syntaxes."

> What actually happened (e.g. what warnings or errors did you get)?

 "The following warnings were flagged:"

```
Unexpected unknown property "..."
```

**Comment by jeddy3** [comment]
@MrVladevoit Thanks for the report and for using the template.

I can reproduce this using [the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIACBbAhgB3gAmNg1jAJ44BuaANgK4zYC+IANCAM5EB2ALmgB4IgUYLFk3AR2AMwCWAc37AAOu2zYFIAE5VBLNTkXKVqkBnUQMMdZyIBadhGtV2+OwHd2u7J00wlKukoY6IA) with:

```less
@map: { key: value }
```

We'll need to detect this construct in the [`isStandardSyntaxDeclaration` util](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxDeclaration.js), similarly to how we detected nested SCSS properties, and return false for it.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
